packages/amplify_datastore/android/src/test/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePluginTest.kt

```diff
@@ -197,6 +203,114 @@ class AmplifyDataStorePluginTest {
             )
     }
 
+    @Test
+    fun test_Save_Success_Result_No_Predicates() {
+        val testRequest: HashMap<String, Any> = (readMapFromFile("save_api",
+                "request/instance_without_predicate.json",
+                HashMap::class.java) as HashMap<String, Any>)
+
+        val serializedModelData: HashMap<String, Any> = testRequest["serializedModel"] as HashMap<String, Any>
+
+        val serializedModel = SerializedModel.builder()
+                .serializedData(serializedModelData)
+                .modelSchema(modelSchema)
+                .build()
+
+        val dataStoreItemChange = DataStoreItemChange.builder<SerializedModel>()
+                .item(serializedModel)
+                .initiator(DataStoreItemChange.Initiator.LOCAL)
+                .itemClass(SerializedModel::class.java)
+                .type(DataStoreItemChange.Type.CREATE)
+                .randomUuid()
+                .build()
+
+        doAnswer { invocation: InvocationOnMock ->
+            (invocation.arguments[2] as Consumer<DataStoreItemChange<SerializedModel>>).accept(
+                    dataStoreItemChange)
+            null as Void?
+        }.`when`(mockAmplifyDataStorePlugin).save(
```

Comment:
We should assertions for arguments 0 and 1 that we correctly deserialized serializedModel and queryPredicates, otherwise we are completely missing that logic in testing. (Or use Argument captures as Jameson suggested in the query PR https://github.com/aws-amplify/amplify-flutter/pull/158#discussion_r510537798)